solidity/ExpirableAirdrop.sol

```diff
@@ -1,60 +1,111 @@
   error Expired();
+  error NotExpired();
   error NotInMerkle();
 
-  event Claimed(address indexed to, uint256 amount);
+  event Claimed(address indexed _claimee, address indexed _receiver, uint256 _amount);
+  event Deposited(uint256 _amount);
+  event Retrieved(address _receiver);
 
   /// @notice Creates a new ExpirableAirdrop contract
+  /// @param _govAddy governor address
   /// @param _token token to airdrop
+  /// @param _expirationTimestamp expiration timestamp
   /// @param _merkleRoot of claimees
   constructor(
+    address _govAddy,
     IERC20 _token,
     uint256 _expirationTimestamp,
     bytes32 _merkleRoot
-  ) {
+  ) Governable(_govAddy) {
     token = _token;
     expirationTimestamp = _expirationTimestamp;
     merkleRoot = _merkleRoot;
   }
 
-  /// @notice Allows claiming tokens if address is part of merkle tree
-  /// @param to address of claimee
-  /// @param amount of tokens owed to claimee
-  /// @param proof merkle proof to prove address and amount are in tree
-  function claim(
-    address to,
-    uint256 amount,
-    bytes32[] calldata proof
-  ) external {
+  function _claim(
+    address _claimee,
+    address _receiver,
+    uint256 _amount,
+    bytes32[] calldata _proof
+  ) internal {
     // CHECKS
     if (block.timestamp > expirationTimestamp) revert Expired();
-    if (hasClaimed[to]) revert AlreadyClaimed();
-    bytes32 leaf = keccak256(abi.encodePacked(to, amount));
-    bool isValidLeaf = MerkleProof.verify(proof, merkleRoot, leaf);
-    if (!isValidLeaf) revert NotInMerkle();
+    if (hasClaimed[_claimee]) revert AlreadyClaimed();
+    bytes32 _leaf = keccak256(abi.encodePacked(_claimee, _amount));
+    bool _isValidLeaf = MerkleProof.verify(_proof, merkleRoot, _leaf);
+    if (!_isValidLeaf) revert NotInMerkle();
 
     // EFFECTS
-    hasClaimed[to] = true;
+    hasClaimed[_claimee] = true;
+
+    // INTERACTIONS
+    token.safeTransfer(_receiver, _amount);
+
+    emit Claimed(_claimee, _receiver, _amount);
+  }
+
+  /// @notice Allows claiming tokens, if address is part of merkle tree
+  /// @param _claimee address of claimee
+  /// @param _amount of tokens owed to claimee
+  /// @param _proof merkle proof to prove address and amount are in tree
+  function claim(
+    address _claimee,
+    uint256 _amount,
+    bytes32[] calldata _proof
+  ) external {
+    _claim(_claimee, _claimee, _amount, _proof);
+  }
+
+  /// @notice Allows claiming tokens and send tokens to a receiver, if address is part of merkle tree
+  /// @param _receiver address to send tokens
+  /// @param _amount of tokens owed to claimee
+  /// @param _proof merkle proof to prove address and amount are in tree
+  function claimAndTransfer(
+    address _receiver,
+    uint256 _amount,
+    bytes32[] calldata _proof
+  ) external {
+    _claim(msg.sender, _receiver, _amount, _proof);
+  }
+
+  /// @notice Deposit tokens to airdrop
+  /// @param _amount of tokens to deposit
+  function depositTokens(uint256 _amount) external {
```

Comment:
I was going to argue this was not entirely necessary, but then I saw you guys comments. Seems sensible to have the `depositTokens` function, specially on Optimism that everything is pretty cheap.

Reply:
Yeah, agree. Makes sense to keep it, and we can decide if we want to use it or not later

Reply:
Cool, I'll keep it